packages/process/src/node/process.ts

```diff
@@ -172,4 +178,34 @@ export abstract class Process {
     protected isForkOptions(options: any): options is ForkOptions {
         return !!options && !!options.modulePath;
     }
+
+    protected readonly initialCwd: string;
+
+    /**
+     * @returns the current working directory as a URI (usually file:// URI)
+     */
+    public getCwdURI(): Promise<string> {
+        if (isOSX) {
+            return new Promise<string>(resolve => {
+                exec('lsof -p ' + this.pid + ' | grep cwd', (error, stdout, stderr) => {
+                    if (stdout !== '') {
+                        resolve(FileUri.create(stdout.substring(stdout.indexOf('/'), stdout.length - 1)).toString());
+                    }
+                });
+            });
+        } else if (isWindows) {
+            return new Promise<string>(resolve => {
+                resolve(FileUri.create(this.initialCwd).toString());
```

Comment:
I added the fall back for OSX. injection ApplicationPackage for falling back to `projectPath`  doesn't  seem to be right. As `spawn` without cwd will use the cwd of the parent process if I'm not mistaken. Also it would add another dependency that is never really used as all clients actually pass a cwd. I'd rather make it mandatory.